### PR TITLE
MGMT-11274: Add Packer and Terraform files to run AI test machine

### PR DESCRIPTION
**Pull request by vrutkovs**
This adds Packer files to prepare CentOS8-based image to run minikube with assisted-service.
The second commit adds Terraform files to start this machine in Nutanix cluster.

Ref: https://github.com/openshift/release/pull/31726

### CODE REVIEW

**Review by adriengentil — APPROVED** [review]

File: packer_files/nutanix_centos_template/centos-config/ks.cfg [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+logvol / --vgname=vg00 --size=5000 --grow --name=root --fstype=ext4
+
+# Partition clearing information
+clearpart --none --initlabel
+
+# System timezone
+timezone America/New_York --isUtc
+
+# Root password
+sshkey --username=root "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQC8L9EzpTcR0VZTf2Pqt/qrdtIeMZoqm2a+KvVHG7XdCDj4DW3d/8bLPduIm6GmvhU0F09H1VYfTvAviB5CMM7FVmfVEH112wZhUVjpcuPf75/1on9xernbYiysqKmc8pPaadlko7NQfFc36LrV6UpUpZoFUqd7FdZLbi6hLT41tQqRiNto+TbyfxtusyB/eynrxLaRVPA5+Jmah4jir9L0j7m2e9bn3om+rXBPY1zlp5pds+xYwfkRWaXJ93G2s1QlpicNBJOckAVgEOZddHDGjGqtFCt7v58nEF53dgpdDk26qKeqiavRdQ5NDRXMNPL/YlqVz/EG4HBtL628BvKCVuok910olIqEr8sQK5WTla1WMmiEjByYX2w6tSoAgvmfwczshko0Qq8h4oX/rmn4TDjpgnXSQvaCaq4IePcANrd7wQSgSQmXnglP2HuotLhE3+/FVXdu39/ZJDAHk9G3ACa2JxgMZTe6UlfI9uAP4yCOpvhspvLadq5GTzDysabFpuWli5F5BEpLjalt+HHBKnQzFtjthY8qZ+Ey71Z1XK/f7H+kQ7D22H08AQu7ccmana7bkT6wsauRdfojyveMmapI8DL8iit7+liX/n6+58wSrpjsozHetaQ0dQjAYUdqYDFs4YZBMomoL4xFaBpLUuVqmVwrzbgRNVX7miu1uw== vrutkovs@redhat.com"
```
Comment (eliorerz) [inline packer_files/nutanix_centos_template/centos-config/ks.cfg:45]: maybe we should do the same as we are doing on vsphere:

```suggestion
sshkey --username=root "${key}"
```

And set `key` on `sources.pkr.hcl`

File: packer_files/nutanix_centos_template/sources.pkr.hcl [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    image_type = "ISO_IMAGE"
+    source_image_name = var.centos_iso_image_name
+  }
+
+  cd_files = ["centos-config/ks.cfg"]
+  cd_label = "OEMDRV"
+
+  vm_disks {
+    image_type = "DISK"
+    disk_size_gb = 40
```
Comment (eliorerz) [inline packer_files/nutanix_centos_template/sources.pkr.hcl:20]:
```suggestion
    disk_size_gb = var.disk_size / 1024
```

File: packer_files/nutanix_centos_template/variables.pkr.hcl [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+variable "image_name" {
+  type = string
+  default = "test"
+  description = "Image name"
+}
+
+variable "disk_size" {
+  type = number
+  default = 240000
+  description = "The VM disk size in MB. default 80G"
```
Comment (eliorerz) [inline packer_files/nutanix_centos_template/variables.pkr.hcl:47]:
```suggestion
  description = "The VM disk size in MB. default 240G"
```

File: packer_files/nutanix_centos_template/variables.pkr.hcl [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+variable "memory_size" {
+  type = string
+  default = "16984"
+  description = "The VM RAM size in MB. TODO: Adjust this variable to the right size"
```
Comment (eliorerz) [inline packer_files/nutanix_centos_template/variables.pkr.hcl:53]:
```suggestion
  default = "16384"
  description = "The VM RAM size in MB"
```

File: packer_files/nutanix_centos_template/variables.pkr.hcl [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+variable "vcpus" {
+  type = string
+  default = "4"
+  description = "The num of CPUs for this VM. TODO: Adjust this variable to the right size"
```
Comment (eliorerz) [inline packer_files/nutanix_centos_template/variables.pkr.hcl:59]:
```suggestion
  description = "The num of CPUs for this VM"
```

File: terraform_files/nutanix-ci-machine/variables-nutanix.tf [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+variable "build_id" {
+  type        = string
+  description = "The CI build id"
+}
+
+variable "memory" {
+  type = number
+  default = 8192
```
Comment (eliorerz) [inline terraform_files/nutanix-ci-machine/variables-nutanix.tf:48]: I don't think we should have default value for memory in the terraform variables file. 
Set it dynamically on tf apply

File: terraform_files/nutanix-ci-machine/variables-nutanix.tf [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+  description = "RAM in MiB allocated to masters"
+}
+
+variable "vcpu" {
+  type = number
+  default = 2
```
Comment (eliorerz) [inline terraform_files/nutanix-ci-machine/variables-nutanix.tf:54]: It doesn't seem like enough, maybe change it to 4/8

```suggestion
  default = 4
```

**Review by eliorerz — CHANGES_REQUESTED** [review]

File: packer_files/nutanix_centos_template/centos-config/ks.cfg (hunk 1 above)
Comment (vrutkovs) [reply packer_files/nutanix_centos_template/centos-config/ks.cfg:45]: Nutanix terraform won't let me create a file on CD-ROM from (templated) contents. I think I can fix it by creating a local file, but  that's something I could fix in the follow up PR

File: packer_files/nutanix_centos_template/centos-config/ks.cfg [hunk 8]
```diff
@@ -0,0 +1,58 @@
+# Use graphical install
```
[the remark this answers is not in the document]
Comment (vrutkovs) [reply packer_files/nutanix_centos_template/centos-config/ks.cfg:1]: Nutanix terraform provider won't let me create new files from (templated) content - and apparently it cannot be renamed too.

I'll investigate if I can create a local file with terraform, template it and rename it

**Review by eliorerz — APPROVED** [review]

**Pull request merged**